SQLite: Document Command Timeout connection string keyword (#25154)
dotnet/docs: docs/standard/data/sqlite/connection-strings.md

---
title: Connection strings

> ms.date: 07/14/2021

Reviewer: ```suggestion
ms.date: 07/14/2021
no-loc: Command Timeout,Default Timeout,Data Source,Recursive Triggers
```

The following connection string keywords can be used with Microsoft.Data.Sqlite:

> ### Data Source

Reviewer: ```suggestion
### Data source
```

Per our style guides, these were correct as is. Titles and heading should be in [sentence case](https://docs.microsoft.com/en-us/contribute/style-quick-start#other-style-and-voice-issues-to-watch-for). However, if these are proper names that can be capitalized, but I do not believe that these are.

Author: But... they keywords are case-sensitive. I know this goes against casing guidelines, but I think it's important preserve the actual case.

Reviewer: Ah, good to know. I was unaware of that.

There's no need enable foreign keys if, like in e_sqlite3, SQLITE_DEFAULT_FOREIGN_KEYS was used to compile the native
SQLite library.

> ### Recursive Triggers

Reviewer: ```suggestion
### Recursive triggers
```

| True  | Sends `PRAGMA recursive_triggers` immediately after opening the connection. |
| False | Doesn't send `PRAGMA recursive_triggers`. This is the default.              |

> ### Default Timeout

Reviewer: ```suggestion
### Default timeout
```

> The default timeout (in seconds) for executing commands. The default value is 30. *Command Timeout* is an alias of this keyword.

Reviewer: ```suggestion
The default timeout (in seconds) for executing commands. The default value is 30. The*command timeout* is an alias of this keyword.
```

I believe that we're confusing the way to show these keywords, or aliases, or property names. If we're referring to it conceptually, we should call it "default timeout", or as it's property name `DefaultTimeout`. It shouldn't be *Default Timeout".

Author: No properties here. Connection strings only have keywords. Some keywords (like `Command Timeout`) are merely aliases for the canonical form of the keyword (in this case, `Default Timeout`)

Reviewer: Ok, in that case we need to add:

```markdown
no-loc: Command Timeout, Default Timeout, Data Source, etc. 
```

To the docs' metadata, it wasn't obvious that these were keywords.

This value can be overridden using <xref:Microsoft.Data.Sqlite.SqliteConnection.DefaultTimeout> which in turn can be overridden using <xref:Microsoft.Data.Sqlite.SqliteCommand.CommandTimeout>.

> [!NOTE]

> > The Default Timeout keyword was added in version 6.0.

Reviewer: ```suggestion
> The `DefaultTimeout` property was added in version 6.0.
```

Instead of an alert here at the end, you might want to open this section with this info. For example, "Starting with version 6.0, you can use the `DefaultTimeout` to ..."

Author: Again, there are no properties in connection strings--only on .NET types.

Using `NOTE` is consistent with other keywords on this page, and with the EF Core docs. Given that we have three supported versions, I think this information is important to see while scanning.
